src/openms/source/FORMAT/FASTAFile.cpp

```diff
@@ -39,184 +39,242 @@
+        {
+            throw Exception::FileNotReadable(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, filename);
+        }
+
+        if (infile_.is_open()) infile_.close(); // precaution
+
+        infile_.open(filename.c_str(), std::ios::binary | std::ios::in);
+        infile_.seekg(0, infile_.end);
+        fileSize_ = infile_.tellg();
+        infile_.seekg(0, infile_.beg);
+
+        std::streambuf* sb = infile_.rdbuf();
+        while(sb->sgetc() == '#')// Skip the header of PEFF files (http://www.psidev.info/peff)
+        {
+            infile_.ignore(numeric_limits<streamsize>::max(),'\n');
+        }
+        entries_read_ = 0;
     }
-    ++entries_read_;
-    s.removeWhitespaces();
-    protein.sequence = s; // assign here, since 's' might have higher capacity, thus wasting memory (usually 10-15%)
-
-    // handle id
-    id.trim();
-    String::size_type position = id.find_first_of(" \v\t");
-    if (position == String::npos)
+
+    bool FASTAFile::readNext(FASTAEntry& protein)
     {
-      protein.identifier = std::move(id);
-      protein.description = "";
+        if (infile_.eof())
+        {
+            return false;
+        }
+
+        String id, s;
+        if (readEntry_(id, s) == false)
+        {
+            if (entries_read_ == 0) s = "The first entry could not be read!";
+            else s = "Only " + String(entries_read_) + " proteins could be read. The record after failed.";
+            throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "", "Error while parsing FASTA file! " + s + " Please check the file!");
+        }
+        ++entries_read_;
+
+        protein.sequence = s; // assign here, since 's' might have higher capacity, thus wasting memory (usually 10-15%)
+        // handle id
+        id.trim();
+        String::size_type position = id.find_first_of(" \v\t");
+        if (position == String::npos)
+        {
+            protein.identifier = std::move(id);
+            protein.description = "";
+        }
+        else
+        {
+            protein.identifier = id.substr(0, position);
+            protein.description = id.suffix(id.size() - position - 1);
+        }
+        return true;
     }
-    else
+
+    std::streampos FASTAFile::position()
     {
-      protein.identifier = id.substr(0, position);
-      protein.description = id.suffix(id.size() - position - 1);
+        return infile_.tellg();
     }
 
-    return true;
-  }
-
-  std::streampos FASTAFile::position() const
-  {
-    return seqan::position(*static_cast<FASTARecordReader*>(reader_.get()));
-  }
-
-  bool FASTAFile::setPosition(const std::streampos& pos)
-  {
-    return (seqan::setPosition(*static_cast<FASTARecordReader*>(reader_.get()), pos) == 0);
-  }
-
-  bool FASTAFile::atEnd() const
-  {
-    return seqan::atEnd(*static_cast<FASTARecordReader*>(reader_.get()));
-  }
-
-  void FASTAFile::load(const String& filename, vector<FASTAEntry>& data) const
-  {
-    startProgress(0, 1, "Loading FASTA file");
```

Comment:
you need to undo the changes here (and all other occurrences of `startProgress`, `nextProgress` and `endProgress` which got deleted.